src/worker.rs

```diff
@@ -8,15 +8,47 @@ use std::{thread, time};
 
 use camunda_client::models::VariableValueDto;
 
-type VariablesMap = HashMap<String, VariableValueDto>;
+pub type VariablesMap = HashMap<String, VariableValueDto>;
+pub type HandlerResult = Result<VariablesMap, Box<dyn Error>>;
 
 pub fn wait(interval_seconds: u64) {
     thread::sleep(time::Duration::from_secs(interval_seconds));
 }
 
+pub struct Handler {
+    pub topic: String,
+    pub worker_id: String,
+    pub config: crate::config::Config,
+    pub action: Box<dyn Fn(Task) -> HandlerResult + Send + 'static>,
+}
```

Comment:
This should only contain `topic` and `action` (which needs renaming)

the worker id must stay the same across handlers, and I see no particular advantage in splitting the config

this will then allow to use the Handler struct for the `run_topic_handler` function as well

Reply:
Good point.

What threw me off about the config was that it contained `topic` and `worker_id`, which I just realized were not used.

What would you suggest as a better name for `action`?

Reply:
`handler_function` for example.

not sure what makes you think that `topic` and `worker_id` are not used. They definitely need to be included in the config as they are a fundamental part of the configuration of the worker and as such they need to be included in the standard config for the library.

As above, the `worker_id` **must** not change for a worker that handles multiple topics, and in regards to the `topic` config variable, rather than removing it from the config, I would rather change it into a map. (see [this answer](https://stackoverflow.com/questions/41242836/how-to-define-a-map-in-toml)

Reply:
> not sure what makes you think that `topic` and `worker_id` are not used.

The fact that neither of the variables was used anywhere in the whole project, at least not as part of the `Config` struct.

I agree that `worker_id` makes sense as a member of `Config` rather than a separate argument to `run_topic_handler(s)`. I've moved it and also renamed `action` to `handler_function` as you suggested.

I am, howewer, not clear on how you expect `topic` to work. You're saying it should be a map. What would each key and value represent?

Reply:
I am talking of something like the following

```topics = { some_topic = "TopicString", some_other_topic = "OtherTopicString" }```

where the keys of the map are to be used in the code as handles for the topic strings, which are in the values.

Reply:
The fact that worker_id and topic were not used in this library does not mean they are not used. This is a module meant to be a utility for users of the library in order to parse common worker configuration parameters. They can then wire the result of that parsing into a handler, or use anything else like hardcoded strings during prototyping, or anything really.

Reply:
I don't think adding extra stuff to the `Config` is a good idea.

Firstly, IMHO, it makes the library more confusing - I would not expect a library to define configuration variables which end up being unused internally, I don't think I've ever seen a library do that.

Secondly, I think it makes it harder to use. You can't predict what the needs of the user are, and it's very likely that they will have to load their own config with their own application-specific variables. It's easier to `impl From<MyCustomConfig> for camunda_worker::config::Config` or even construct it manually before calling `run_topic_handler` if there's less stuff to fill.

I agree that `topic`, for example, is something that the user needs to have, but I'd rather have him/her wire the needed variables him/herself as opposed to dealing with a map or a list of maps, which are, IMHO, slightly confusing in this situation to begin with.

I like the idea of specifying `wait_interval_seconds` per-topic though, but I'd suggest to have it be a part of the `Handler` struct instead (and delete it from `Config`), so that the connection between related things is more obvious - you would be specifying a function to call, the topic it's dealing with and its polling interval all in one place.

Reply:
I disagree. 

The topic is part of the fundamental configuration of the worker, so it is hardly "extra stuff" to be added to the config.

Please try the implementation with the config looking like the following 

```
[[topic]]
name = "validate_business_key"
engine_string = "ValidateBusinessKey"
wait_interval_seconds = 60

[[topic]]
name = "validate_order_id"
engine_string = "ValidateOrderNumber"
wait_interval_seconds = 600

[...]
```

I prefer the explicit approach of providing the `Config` module to the users so that the obvious configurations are "batteries included", with the side effect of being well documented, but by not implicitly wiring such configurations into the code - and rather having the users fill parameters to the topic handlers with those convenience functions ( EG: `worker::run_topic_handler( config::topics.validate_business_key [..]`)

This way, for example during prototyping, they are free to hardcode values.. or even use a function to determine the wait interval for their topics.

In essence, it's much more extensible and convenient in the long run to have the values parsed by the library via the config module and wired in wherever needed by the consumer code explicitly.

Reply:
I apologize, maybe I'm having one of those days when my brain refuses to start, but your comment confuses me.

First, you seem to be working off the assumption that the library supports loading the `Config` from `TOML`, which it currently does not. Should it be added? (I'd prefer if it wasn't, config loading should IMHO be handled by a library chosen by the user)

> I prefer the explicit approach of providing the `Config` module to the users so that the obvious configurations are "batteries included", with the side effect of being well documented, but by not implicitly wiring such configurations into the code - and rather having the users fill parameters to the topic handlers with those convenience functions ( EG: `worker::run_topic_handler( config::topics.validate_business_key [..]`)

Which convenience functions are you talking about? What is `config::topics.validate_business_key`? Is it a function in the `config` module? How can it exist if `validate_business_key` is specified in the config file by a user?

> This way, for example during prototyping, they are free to hardcode values.. or even use a function to determine the wait interval for their topics.

They are free to do that with what I'm proposing, too.

I'd love to try and implement your suggestion, but at this point, I really don't know what it is that you're suggesting. Maybe you could reword a bit?

Reply:
Apologies, I should have specified that there are other planned changes that interact with and build upon what I am talking about above.

For now handling the topics in the config as a list of maps will be fine. In the future the Config will be automatically parsed from file, and utility macros can be built so that the maps is automatically transformed in a struct. At least, that's my plan.

Please, once again, revert your changes to the config struct in regards to the topic field and squash the commits so that this PR can be merged.

Reply:
I have squashed and pushed as you said, hopefully it's close to what you've envisioned. Since I'm going offline in a few moments, please feel free to further adjust the code to your liking (if needed) and merge, unless there are bigger changes to do, in which case I will take a look at them on Monday :slightly_smiling_face: